wordpress-mobile/WordPress-Android
# Migrate GCM Registration Service from JobIntentService to WorkManager

Fixes #16015 

This PR migrated JobIntentService to WorkManager for GCM token registration at the startup, and authn stages

JobIntentService is deprecated, and known to cause crashes on low end devices, on below Android 10.  

-----

## To Test:
Test 1
- install WP/JP app
- Login 
- Verify, it works as before

Test 2
- Logout
- Login again
- Verify, works as before

Test 3
- Delete the app
- Fresh install the app
- Go through signup
- Ensure, everything works as before

Test 4
- If you have a low end device with Android 10 or below, try reproducing the crash in the issue with current version, and then test with this version to see if it fixes the issue




-----

## Regression Notes

1. Potential unintended areas of impact

    - Login/Signup

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Tested login, signup, and fresh install

3. What automated tests I added (or what prevented me from doing so)

    - Existing unit tests

-----

## PR Submission Checklist:
